config/checkstyle-non-main-files-suppressions.xml

```diff
@@ -167,4 +167,94 @@
   <!-- We use negative suppression to check workflow files only. -->
   <suppress id="workflowJobStepSpacing" files="^.((?!github[\\/]workflows[\\/]).)*$" />
 
+  <!-- Examples macro suppressions -->
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]annotation[\\/]annotationlocation.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]annotation[\\/]annotationonsameline.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]annotation[\\/]annotationusestyle.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]annotation[\\/]missingdeprecated.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]annotation[\\/]missingoverride.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]annotation[\\/]suppresswarnings.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]blocks[\\/]avoidnestedblocks.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]blocks[\\/]emptyblock.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]blocks[\\/]emptycatchblock.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]blocks[\\/]leftcurly.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]blocks[\\/]needbraces.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]blocks[\\/]rightcurly.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]design[\\/]designforextension.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]design[\\/]finalclass.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]design[\\/]hideutilityclassconstructor.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]imports[\\/]importcontrol.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]filters[\\/]suppressionfilter.xml.template"/>
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]filters[\\/]suppressionxpathfilter.xml.template"/>
+
+  <!-- Parent module macro suppressions -->
+  <suppress id="parentModuleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]coding[\\/]unnecessarysemicolonafteroutertypedeclaration.xml.template"/>
+  <suppress id="parentModuleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]coding[\\/]unnecessarysemicolonaftertypememberdeclaration.xml.template"/>
+
+  <!-- Violations messages macro suppressions -->
```

Comment:
Obviously some of these are still in the works (suppressions and filters), but it looks like we missed some checks here.